### ISSUE TITLE
Microsoft.KeyVault/vaults/secrets being forced to recreate

### ISSUE THREAD

**Issue by luifa7** [Azure/terraform-provider-azapi#538]
Existing resource created with azure/azapi v1.12.1 had no explicit location in the definition, so it was automatically set to westeurope.
On upgrading to azure/azapi v1.13.1, the location is now set  to null (like i the bugfixes described). This triggers a force recreation.

Setting location for Microsoft.KeyVault/vaults/secrets leads to schema validation error, the only workaround is to set

```
location                  = "westeurope"
schema_validation_enabled = false
```

Versions:

```
Terraform v1.8.5
provider registry.terraform.io/azure/azapi v1.13.1
```

**Comment by ms-henglu** [comment]
Hi @luifa7 ,

Thank you for taking time to report this issue and apologize for late response.

This seems to be a mismatch between the API and swagger definition, I've opened this issue to track it: https://github.com/Azure/azure-rest-api-specs/issues/30259

As a workaround, please add the location to the config and disable the schema validation.

**Comment by egorshulga** [comment]
It got worse, and the workaround does not work anymore😕

```log
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).PlanResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-azapi_v2.0.1.exe plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x6 pc=0x5052f2]

goroutine 90 [running]:
math/big.(*Float).Float64(0xcaa660?)
        math/big/float.go:966 +0x12
github.com/Azure/terraform-provider-azapi/internal/services/dynamic.attrValueToJSON({0xfa9dca0, 0xc000909fe0?}, 0x0)
        github.com/Azure/terraform-provider-azapi/internal/services/dynamic/dynamic.go:67 +0x1dc
github.com/Azure/terraform-provider-azapi/internal/services/dynamic.attrMapToJSON(0xc0007f9b20?, 0xc0007f9a50?)
        github.com/Azure/terraform-provider-azapi/internal/services/dynamic/dynamic.go:39 +0xb0
github.com/Azure/terraform-provider-azapi/internal/services/dynamic.attrValueToJSON({0xfa9d098, 0xc001848810?}, 0x0)
        github.com/Azure/terraform-provider-azapi/internal/services/dynamic/dynamic.go:94 +0x90f
github.com/Azure/terraform-provider-azapi/internal/services/dynamic.attrMapToJSON(0xc0007fa0f8?, 0xc0007fa028?)
        github.com/Azure/terraform-provider-azapi/internal/services/dynamic/dynamic.go:39 +0xb0
github.com/Azure/terraform-provider-azapi/internal/services/dynamic.attrValueToJSON({0xfa9d098, 0xc001848828?}, 0x0)
        github.com/Azure/terraform-provider-azapi/internal/services/dynamic/dynamic.go:94 +0x90f
github.com/Azure/terraform-provider-azapi/internal/services/dynamic.attrMapToJSON(0xc0007fa6d0?, 0xc0007fa600?)
        github.com/Azure/terraform-provider-azapi/internal/services/dynamic/dynamic.go:39 +0xb0
github.com/Azure/terraform-provider-azapi/internal/services/dynamic.attrValueToJSON({0xfa9d098, 0xc001848840?}, 0x0)
        github.com/Azure/terraform-provider-azapi/internal/services/dynamic/dynamic.go:94 +0x90f
github.com/Azure/terraform-provider-azapi/internal/services/dynamic.ToJSON(...)
        github.com/Azure/terraform-provider-azapi/internal/services/dynamic/dynamic.go:17
github.com/Azure/terraform-provider-azapi/internal/services/dynamic.SemanticallyEqual({{0xfa9d098?, 0xc001848840?}, 0x0?}, {{0xfa9d098?, 0xc0018489f0?}, 0x0?})
        github.com/Azure/terraform-provider-azapi/internal/services/dynamic/dynamic.go:367 +0x52
github.com/Azure/terraform-provider-azapi/internal/services/myplanmodifier.dynamicUseStateWhen.PlanModifyDynamic({_}, {_, _}, {{{0xc000909f70, 0x1, 0x1}}, {0x0, {0x0, 0x0, 0x0}}, ...}, ...)
        github.com/Azure/terraform-provider-azapi/internal/services/myplanmodifier/dynamic_use_state_when.go:37 +0xf3
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.AttributePlanModifyDynamic({_, _}, {_, _}, {{{0xc000909f70, 0x1, 0x1}}, {0x0, {0x0, 0x0, ...}}, ...}, ...)
        github.com/hashicorp/terraform-plugin-framework@v1.11.0/internal/fwserver/attribute_plan_modification.go:2567 +0xa56
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.AttributeModifyPlan({_, _}, {_, _}, {{{0xc000909f70, 0x1, 0x1}}, {0x0, {0x0, 0x0, ...}}, ...}, ...)
        github.com/hashicorp/terraform-plugin-framework@v1.11.0/internal/fwserver/attribute_plan_modification.go:116 +0x9fb
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.SchemaModifyPlan({0xfa9b218, 0xc001d68c90}, {0xfaa31b0, 0xc000114be0}, {{{{0xfaa06c0, 0xc001e94ff0}, {0xcf65c0, 0xc001d73d40}}, {0xfaa31b0, 0xc000114be0}}, ...}, ...)
        github.com/hashicorp/terraform-plugin-framework@v1.11.0/internal/fwserver/schema_plan_modification.go:123 +0x825
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).PlanResourceChange(0xc0002641e0, {0xfa9b218, 0xc001d68c90}, 0xc00042b560, 0xc0007fd6b0)
        github.com/hashicorp/terraform-plugin-framework@v1.11.0/internal/fwserver/server_planresourcechange.go:260 +0x1013
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).PlanResourceChange(0xc0002641e0, {0xfa9b218?, 0xc001d68b70?}, 0xc001b48050)
        github.com/hashicorp/terraform-plugin-framework@v1.11.0/internal/proto6server/server_planresourcechange.go:64 +0x4e5
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).PlanResourceChange(0xc0002672c0, {0xfa9b218?, 0xc001d57d10?}, 0xc000e08500)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov6/tf6server/server.go:825 +0x3f0
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_PlanResourceChange_Handler({0xdfc020?, 0xc0002672c0}, {0xfa9b218, 0xc001d57d10}, 0xc000e08480, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:500 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001a7000, {0xfa9b218, 0xc001d57c50}, {0xfaa1438, 0xc000174000}, 0xc001e96c60, 0xc000378e10, 0x10101c60, 0x0)
        google.golang.org/grpc@v1.65.0/server.go:1379 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc0001a7000, {0xfaa1438, 0xc000174000}, 0xc001e96c60)
        google.golang.org/grpc@v1.65.0/server.go:1790 +0x1016
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.65.0/server.go:1029 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 36
        google.golang.org/grpc@v1.65.0/server.go:1040 +0x135

Error: The terraform-provider-azapi_v2.0.1.exe plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

I guess I would need to switch to the approach with custom scripts for creating the secrets we need...

**Comment by ms-henglu** [comment]
Hi @egorshulga , thanks for reporting this issue, I opened this PR to fix it: https://github.com/Azure/terraform-provider-azapi/pull/648

The crash happens when the `body` contains an unknown float value. Please try to remove the unknown float value in the `body` to mitigate the issue.

**Comment by ms-henglu** [comment]
Hi @egorshulga , above fix has been released in v2.1.0, please give it a try.

**Comment by ms-henglu** [comment]
I'll close this issue in favor of https://github.com/Azure/terraform-provider-azapi/issues/655 which also includes other resource types that have same issue.

Please subscribe to https://github.com/Azure/terraform-provider-azapi/issues/655 for any updates. Thanks.
